src/main/scala/com/fulcrumgenomics/fastq/FastqToBam.scala

```diff
@@ -135,6 +136,7 @@ class FastqToBam
     val rg = new SAMReadGroupRecord(this.readGroupId)
     rg.setSample(sample)
     rg.setLibrary(library)
+    this.barcode.foreach(bc => rg.setBarcodes(util.Arrays.asList(bc)))
```

Comment:
not in an idea, but this could work:
```
    this.barcode.foreach(bc => rg.setBarcodes(util.Arrays.asList(bc)))
    if (this.barcode.nonEmpty) rg.setBarcodes(...)
```